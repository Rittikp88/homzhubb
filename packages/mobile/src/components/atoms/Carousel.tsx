import React from 'react';
import Carousel from 'react-native-snap-carousel';
import { StyleProp, ViewStyle } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';

interface ICarouselProps<T> {
  carouselData: T[];
  carouselItem: (item: T) => React.ReactElement;
  activeSlide: number;
  currentSlide: (index: number) => void;
  bubbleRef: (ref: any) => void;
  contentStyle?: StyleProp<ViewStyle>;
}

export class SnapCarousel<T> extends React.PureComponent<ICarouselProps<T>> {
  private carouselRef: any;

  public render = (): React.ReactElement => {
    const { carouselData, activeSlide, contentStyle } = this.props;
    return (
      <Carousel
        onLayout={this.updateRef}
        data={carouselData}
        firstItem={activeSlide}
        sliderWidth={theme.viewport.width}
        itemWidth={theme.viewport.width - 20}
        renderItem={this.renderItem}
        activeSlideOffset={20}
        onSnapToItem={this.updateSlideIndex}
        contentContainerCustomStyle={contentStyle}
        ref={(c: any): void => {
          this.carouselRef = c;
        }}
      />
    );
  };

  public renderItem = ({ item }: { item: T }): React.ReactElement => {
    const { carouselItem } = this.props;
    return carouselItem(item);
  };

  public updateSlideIndex = (index: number): void => {
    const { currentSlide } = this.props;
    currentSlide(index);
  };

  public updateRef = (): void => {
    const { bubbleRef } = this.props;
    bubbleRef(this.carouselRef);
  };
}
