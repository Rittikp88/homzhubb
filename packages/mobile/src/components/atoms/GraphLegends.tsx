import React from 'react';
import { View, StyleSheet, ViewStyle } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import { Label, PricePerUnit } from '@homzhub/common/src/components';
import { IGeneralLedgerGraphData } from '@homzhub/common/src/domain/models/GeneralLedgers';

interface IProps {
  data: IGeneralLedgerGraphData[];
  direction: 'row' | 'column';
}

const GraphLegends = (props: IProps): React.ReactElement => {
  const { direction, data } = props;
  let directionStyle = {};
  let directionLegendStyle: ViewStyle = styles.legendContainerColumn;

  if (direction === 'row') {
    directionStyle = styles.rowContainer;
    directionLegendStyle = styles.legendContainerRow;
  }

  directionStyle = { ...directionStyle, ...{ flexDirection: direction } };

  return (
    <View style={[styles.container, directionStyle]}>
      {data.map((legend: IGeneralLedgerGraphData, index: number) => (
        <View style={[styles.legendContainer, directionLegendStyle]} key={`legend-${index}`}>
          <View style={[styles.color, { backgroundColor: legend.svg.fill }]} />
          <View>
            <Label type="regular" textType="regular" style={styles.textColor}>
              {legend.title}
            </Label>
            <PricePerUnit price={legend.value} currency="INR" textSizeType="small" textFontWeight="regular" />
          </View>
        </View>
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
  },
  legendContainer: {
    flexDirection: 'row',
  },
  rowContainer: {
    flexWrap: 'wrap',
    marginTop: 16,
    justifyContent: 'center',
    alignItems: 'center',
  },
  legendContainerRow: {
    marginBottom: 16,
    marginHorizontal: 12,
  },
  legendContainerColumn: {
    marginVertical: 16,
  },
  color: {
    marginEnd: 6,
    marginTop: 4,
    ...(theme.circleCSS(12) as object),
  },
  textColor: {
    color: theme.colors.darkTint4,
  },
});

const memoizedComponent = React.memo(GraphLegends);
export { memoizedComponent as GraphLegends };
