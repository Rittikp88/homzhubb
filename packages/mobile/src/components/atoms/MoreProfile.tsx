import React, { Component } from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';
import { StringUtils } from '@homzhub/common/src/utils/StringUtils';
import { StorageKeys, StorageService } from '@homzhub/common/src/services/storage/StorageService';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Text } from '@homzhub/common/src/components';
import { IUser } from '@homzhub/common/src/domain/models/User';

interface IProps {
  onIconPress: () => void;
}

interface IMoreProfileState {
  user: IUser | null;
}

type Props = IProps;

export class MoreProfile extends Component<Props, IMoreProfileState> {
  public state = {
    user: {} as IUser,
  };

  public componentDidMount = async (): Promise<void> => {
    const user: IUser | null = await StorageService.get(StorageKeys.USER);
    this.setState({ user });
  };

  public render(): React.ReactNode {
    return <View style={styles.container}>{this.renderHeader()}</View>;
  }

  private renderHeader = (): React.ReactElement => {
    const { onIconPress } = this.props;
    const { user } = this.state;
    return (
      <View style={styles.headerContainer}>
        <View style={styles.flexRow}>
          <View style={styles.initialsContainer}>
            <Text type="small" textType="regular" style={styles.initials}>
              {StringUtils.getInitials(user?.full_name ?? 'User')}
            </Text>
          </View>
          <View style={styles.nameContainer}>
            <Text type="small" textType="regular">
              Hello!
            </Text>
            <Text type="regular" textType="semiBold">
              {user?.full_name ?? 'User'}
            </Text>
          </View>
        </View>
        <TouchableOpacity onPress={onIconPress}>
          <Icon name={icons.rightArrow} size={18} color={theme.colors.lowPriority} />
        </TouchableOpacity>
      </View>
    );
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  flexRow: {
    flexDirection: 'row',
  },
  headerContainer: {
    flex: 1,
    backgroundColor: theme.colors.moreSeparator,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: theme.layout.screenPadding,
    height: 120,
  },
  initialsContainer: {
    ...(theme.circleCSS(60) as object),
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: theme.colors.darkTint7,
    borderColor: theme.colors.white,
    borderWidth: 1,
    elevation: 10,
    shadowColor: theme.colors.shadow,
    shadowRadius: 4,
  },
  nameContainer: {
    marginStart: 12,
    justifyContent: 'center',
  },
  initials: {
    color: theme.colors.white,
  },
});
