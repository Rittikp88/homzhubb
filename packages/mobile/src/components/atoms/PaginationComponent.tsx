import React from 'react';
import { Pagination } from 'react-native-snap-carousel';
import { StyleProp, StyleSheet, ViewStyle } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';

interface IPaginationProps {
  dotsLength: number;
  activeSlide: number;
  containerStyle?: StyleProp<ViewStyle>;
  activeDotStyle?: StyleProp<ViewStyle>;
  inactiveDotStyle?: StyleProp<ViewStyle>;
}

export const PaginationComponent = (props: IPaginationProps): React.ReactElement => {
  const { dotsLength, activeSlide, containerStyle, inactiveDotStyle, activeDotStyle } = props;
  return (
    <Pagination
      dotsLength={dotsLength}
      activeDotIndex={activeSlide}
      dotStyle={[styles.dotStyle, activeDotStyle]}
      inactiveDotStyle={[styles.inactiveDotStyle, inactiveDotStyle]}
      inactiveDotOpacity={1}
      inactiveDotScale={1}
      containerStyle={containerStyle}
    />
  );
};

const styles = StyleSheet.create({
  dotStyle: {
    width: 10,
    height: 10,
    borderRadius: 5,
    marginHorizontal: 2,
    borderColor: theme.colors.blue,
    borderStyle: 'solid',
    borderWidth: 3,
    backgroundColor: theme.colors.white,
  },
  inactiveDotStyle: {
    width: 10,
    height: 10,
    borderRadius: 5,
    marginHorizontal: 2,
    backgroundColor: theme.colors.blue,
  },
});
