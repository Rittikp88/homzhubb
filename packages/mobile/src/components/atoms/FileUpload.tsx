import React from 'react';
import { StyleSheet, View } from 'react-native';
import { useTranslation } from 'react-i18next';
import * as RNProgress from 'react-native-progress';
import { theme } from '@homzhub/common/src/styles/theme';
import { viewport } from '@homzhub/common/src/styles/viewport';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Divider } from '@homzhub/common/src/components/atoms/Divider';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import { IDocumentSource } from '@homzhub/mobile/src/components/molecules/UploadBoxComponent';
import { UploadFileType } from '@homzhub/common/src/domain/models/Attachment';

interface IFileUploadProps {
  attachments: IDocumentSource[];
  onDelete: (uri: string) => void;
}

const FileUpload = (props: IFileUploadProps): React.ReactElement | null => {
  const { attachments = [], onDelete } = props;
  const { t } = useTranslation();
  const fileSize = (bytes: number): string => {
    const integral = parseFloat((bytes / (1024 * 1024)).toFixed(1));
    return integral ? `${integral} mb` : `${(bytes / 1024).toFixed(1)} kb`;
  };
  const fileName = (name: string, extension: string): string =>
    name.length > 20 ? `${name.slice(0, 20)}...${extension}` : name;

  if (attachments.length) {
    return (
      <>
        {attachments.map((attachment: IDocumentSource) => {
          const { name, uri, size } = attachment;
          const extension = name.split('.').reverse()[0];
          const fileType = extension === 'pdf' ? UploadFileType.PDF : UploadFileType.IMAGE;
          const isLastAttachment = attachments.indexOf(attachment) === attachments.length - 1;
          const fileIcon = fileType === UploadFileType.PDF ? icons.doc : icons.imageFile;

          const RenderDivider = (): React.ReactElement =>
            !isLastAttachment ? <Divider containerStyles={styles.divider} /> : <View style={styles.endingEmptyView} />;

          const onPress = (): void => onDelete(uri);

          return (
            <>
              <View key={uri} style={styles.fullContainer}>
                <View style={styles.fileIconView}>
                  <Icon name={fileIcon} size={40} color={theme.colors.lowPriority} style={styles.fileIcon} />
                </View>

                <View style={styles.topView}>
                  <View style={styles.titleView}>
                    <Text
                      type="small"
                      textType="semiBold"
                      style={styles.fileName}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                    >
                      {fileName(name, extension)}
                    </Text>
                    <Icon
                      name={icons.close}
                      size={20}
                      color={theme.colors.darkTint3}
                      style={styles.closeIcon}
                      onPress={onPress}
                    />
                  </View>

                  <RNProgress.Bar
                    color="green"
                    progress={1}
                    borderRadius={4}
                    width={viewport.width / 1.5}
                    height={5}
                    unfilledColor={theme.colors.background}
                    style={styles.progressBar}
                  />

                  <View style={styles.bottomView}>
                    <Text type="small" textType="semiBold" style={styles.fileInfo}>
                      {t('common:completed')}
                    </Text>
                    <Text type="small" textType="semiBold" style={styles.fileInfo}>
                      {fileSize(size)}
                    </Text>
                  </View>
                </View>
              </View>
              <RenderDivider />
            </>
          );
        })}
      </>
    );
  }
  return null;
};

export default React.memo(FileUpload);

const styles = StyleSheet.create({
  fullContainer: {
    flex: 1,
    flexDirection: 'row',
    height: 85,
    marginTop: 10,
  },
  fileIconView: {
    flex: 1,
    justifyContent: 'flex-start',
  },
  topView: { flex: 4 },
  titleView: {
    flex: 0.7,
    justifyContent: 'space-between',
    flexDirection: 'row',
  },
  fileName: {
    color: theme.colors.darkTint3,
  },
  closeIcon: {
    marginHorizontal: 10,
  },
  progressBar: {
    borderColor: theme.colors.disabled,
    marginTop: 10,
  },
  bottomView: {
    flex: 1,
    justifyContent: 'space-between',
    flexDirection: 'row',
    marginTop: 8,
  },
  fileInfo: {
    color: theme.colors.darkTint6,
    fontSize: 14,
  },
  divider: {
    borderWidth: 1,
    marginVertical: 5,
    borderColor: theme.colors.divider,
  },
  fileIcon: {
    marginHorizontal: 10,
  },
  endingEmptyView: {
    marginVertical: 5,
  },
});
