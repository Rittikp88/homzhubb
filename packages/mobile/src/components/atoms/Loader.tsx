import React from 'react';
import { ActivityIndicator, StyleSheet, View } from 'react-native';
import { useTranslation } from 'react-i18next';
import { PlatformUtils } from '@homzhub/common/src/utils/PlatformUtils';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { theme } from '@homzhub/common/src/styles/theme';
import { Label } from '@homzhub/common/src/components/atoms/Text';

export const Loader = (): React.ReactElement => {
  const { t } = useTranslation(LocaleConstants.namespacesKey.common);
  return (
    <View style={styles.container}>
      <View style={styles.overlay}>
        <View style={styles.background}>
          <ActivityIndicator
            size={PlatformUtils.isIOS() ? 'large' : 50}
            color={theme.colors.darkTint4}
            style={styles.loader}
          />
          <Label type="regular" textType="regular" style={styles.text}>
            {t('loading')}
          </Label>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: theme.colors.transparent,
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
  },
  overlay: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    justifyContent: 'center',
    alignItems: 'center',
  },
  background: {
    position: 'absolute',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: theme.colors.white,
    borderRadius: 10,
  },
  loader: {
    marginHorizontal: 24,
    marginVertical: 12,
  },
  text: {
    color: theme.colors.darkTint4,
    marginBottom: 12,
  },
});
