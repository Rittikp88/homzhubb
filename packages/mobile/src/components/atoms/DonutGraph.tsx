import React, { useState, useEffect } from 'react';
import { StyleSheet } from 'react-native';
import { PieChart } from 'react-native-svg-charts';
import { theme } from '@homzhub/common/src/styles/theme';
import { EmptyState } from '@homzhub/common/src/components/atoms/EmptyState';
import { GraphLegends } from '@homzhub/mobile/src/components/atoms/GraphLegends';
import { GeneralLedgers, IGeneralLedgerGraphData } from '@homzhub/common/src/domain/models/GeneralLedgers';

const INNER_RADIUS = '70%';
const HEIGHT = 225;
interface IProps {
  data: GeneralLedgers[];
}

const DonutGraph = (props: IProps): React.ReactElement => {
  const { data } = props;
  const [donutData, setDonutData] = useState<IGeneralLedgerGraphData[]>([]);

  useEffect(() => {
    const transformedData: IGeneralLedgerGraphData[] = [];

    data.forEach((ledger: GeneralLedgers, index: number) => {
      const { category, amount } = ledger;
      transformedData.push({
        key: index,
        title: category,
        value: amount,
        svg: { fill: theme.randomHex() },
      });
    });

    setDonutData(transformedData);
  }, [data]);

  const render = (): React.ReactElement => {
    if (data.length === 0) return <EmptyState />;

    return (
      <>
        <PieChart style={styles.pieChart} data={donutData} innerRadius={INNER_RADIUS} />
        <GraphLegends data={donutData} />
      </>
    );
  };
  return render();
};

const styles = StyleSheet.create({
  pieChart: {
    flex: 1,
    marginBottom: 16,
    height: HEIGHT,
  },
});
const memoizedComponent = React.memo(DonutGraph);
export { memoizedComponent as DonutGraph };
