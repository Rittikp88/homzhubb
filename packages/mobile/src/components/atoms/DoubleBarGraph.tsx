import React from 'react';
import { View, StyleSheet, ScrollView } from 'react-native';
import { useSelector } from 'react-redux';
import { BarChart, XAxis, YAxis, Grid } from 'react-native-svg-charts';
import { PathProps } from 'react-native-svg';
import { sum } from 'lodash';
import { theme } from '@homzhub/common/src/styles/theme';
import { UserSelector } from '@homzhub/common/src/modules/user/selectors';
import { EmptyState } from '@homzhub/common/src/components/atoms/EmptyState';
import { GraphLegends } from '@homzhub/mobile/src/components/atoms/GraphLegends';
import { BarGraphLegends, IGeneralLedgerGraphData } from '@homzhub/common/src/domain/models/GeneralLedgers';

interface IProps {
  data: {
    data1: number[];
    data2: number[];
    label: string[];
  };
}

// SVG OPTIONS
const SVG_FONT = { fontSize: 12, fill: theme.colors.darkTint6, fontWeight: '600' as '600' };
const SVG_GRID = { strokeDasharray: [5, 5], stroke: theme.colors.darkTint7, width: '100%' };
// INSET OPTIONS
const VERTICAL_INSET = { top: 8, bottom: 8 };
const HORIZONTAL_INSET = { left: 24, right: 24 };
// VIEWPORT CONSTANTS
const Y_GRID_WIDTH = theme.viewport.width / 7;
const HEIGHT = theme.viewport.height * 0.5;
const WIDTH = theme.viewport.width * 1.75;

const DoubleBarGraph = (props: IProps): React.ReactElement => {
  const {
    data: { data1, data2, label },
  } = props;
  const currency = useSelector(UserSelector.getCurrency);

  const barData = [
    {
      data: data1,
      svg: { fill: theme.colors.expense } as Partial<PathProps>,
    },
    {
      data: data2,
      svg: { fill: theme.colors.income } as Partial<PathProps>,
    },
  ];
  const barGraphLegends = (): IGeneralLedgerGraphData[] => {
    return [
      {
        key: 1,
        title: BarGraphLegends.expense,
        value: sum(data1),
        svg: { fill: theme.colors.expense },
      },
      {
        key: 1,
        title: BarGraphLegends.income,
        value: sum(data2),
        svg: { fill: theme.colors.income },
      },
    ];
  };

  const isData1Empty = data1.every((value) => value === 0);
  const isData2Empty = data2.every((value) => value === 0);
  if (isData1Empty && isData2Empty) {
    return <EmptyState />;
  }

  return (
    <>
      <ScrollView horizontal showsHorizontalScrollIndicator={false}>
        <View style={styles.barGraphContainer}>
          <View style={styles.yContainer}>
            <YAxis
              style={styles.yGrid}
              contentInset={VERTICAL_INSET}
              data={data1.concat(data2)}
              svg={SVG_FONT}
              formatLabel={(value: number): string => `${currency.currencySymbol}${value}`}
            />
            <BarChart
              contentInset={VERTICAL_INSET}
              style={[styles.barGraph, label.length === 1 && styles.derivedWidth]}
              data={barData}
            >
              <Grid direction={Grid.Direction.HORIZONTAL} svg={SVG_GRID} />
            </BarChart>
          </View>
          <XAxis
            style={[styles.xAxis, label.length === 1 && styles.xAxisDerivedWidth]}
            data={label}
            contentInset={HORIZONTAL_INSET}
            svg={SVG_FONT}
            formatLabel={(value, index: number): string => label[index]}
          />
        </View>
      </ScrollView>
      <GraphLegends direction="row" data={barGraphLegends()} />
    </>
  );
};

const memoizedComponent = React.memo(DoubleBarGraph);
export { memoizedComponent as DoubleBarGraph };

const styles = StyleSheet.create({
  barGraphContainer: {
    flexDirection: 'column',
  },
  barGraph: {
    width: WIDTH,
    height: HEIGHT,
  },
  yContainer: {
    flexDirection: 'row',
  },
  yGrid: {
    width: Y_GRID_WIDTH,
    alignItems: 'flex-start',
  },
  xAxis: {
    marginStart: Y_GRID_WIDTH,
    width: WIDTH,
    padding: 12,
  },
  derivedWidth: {
    width: theme.viewport.width * 0.7,
  },
  xAxisDerivedWidth: {
    width: theme.viewport.width * 0.7,
    paddingLeft: theme.viewport.width * 0.275,
  },
});
