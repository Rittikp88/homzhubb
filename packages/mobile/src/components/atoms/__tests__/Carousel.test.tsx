import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { OnboardingData } from '@homzhub/common/src/mocks/onboarding';
import { SnapCarousel } from '@homzhub/mobile/src/components/atoms/Carousel';

const createTestProps = (testProps: any): object => ({
  ...testProps,
});
let props: any;
const mockBubbleRef = jest.fn();
const mockCurrentSlide = jest.fn();

describe('Carousel', () => {
  it('should match snapshot', () => {
    props = createTestProps({
      carouselItems: OnboardingData,
      activeSlide: 0,
      showPagination: true,
      bubbleRef: mockBubbleRef,
      currentSlide: mockCurrentSlide,
    });
    const wrapper = shallow(<SnapCarousel {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
