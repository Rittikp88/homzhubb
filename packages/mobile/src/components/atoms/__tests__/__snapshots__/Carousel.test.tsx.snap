// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should match snapshot 1`] = `
<Fragment>
  <Carousel
    activeAnimationOptions={null}
    activeAnimationType="timing"
    activeSlideAlignment="center"
    activeSlideOffset={20}
    apparitionDelay={0}
    autoplay={false}
    autoplayDelay={1000}
    autoplayInterval={3000}
    callbackOffsetMargin={5}
    containerCustomStyle={Object {}}
    contentContainerCustomStyle={Object {}}
    data={
      Array [
        Object {
          "buttonText": "Next",
          "description": "Sit amet, in enim nunc, etiam amet velit. Malesuada non neque nec in a. Sed massa facilisi vel neque id dolor",
          "heading": "Search",
          "scene": "https://socion-s3-bucket.s3.amazonaws.com/Search_Onboarding.svg",
        },
        Object {
          "buttonText": "Next",
          "description": "Ornare nunc, sagittis est ut. Mi arcu morbi faucibus consectetur auctor. Diam odio sed vel nam ultricies.",
          "heading": "Post",
          "scene": "https://socion-s3-bucket.s3.amazonaws.com/Post_Onboarding.svg",
        },
        Object {
          "buttonText": "Got it",
          "description": "Mattis ut est senectus commodo id sem arcu erat luctus. Id egestas eu enim, varius. Ut tellus id neque enim.",
          "heading": "Maintain",
          "scene": "https://socion-s3-bucket.s3.amazonaws.com/Maintain_Onboarding.svg",
        },
      ]
    }
    enableMomentum={false}
    enableSnap={true}
    firstItem={0}
    hasParallaxImages={false}
    inactiveSlideOpacity={0.7}
    inactiveSlideScale={0.9}
    inactiveSlideShift={0}
    itemWidth={730}
    layout="default"
    lockScrollTimeoutDuration={1000}
    lockScrollWhileSnapping={false}
    loop={false}
    loopClonesPerSide={3}
    onLayout={[Function]}
    onSnapToItem={[Function]}
    renderItem={[Function]}
    scrollEnabled={true}
    shouldOptimizeUpdates={true}
    slideStyle={Object {}}
    sliderWidth={750}
    swipeThreshold={20}
    useScrollView={false}
    vertical={false}
  />
  <Pagination
    activeDotIndex={0}
    animatedDuration={250}
    animatedFriction={4}
    animatedTension={50}
    delayPressInDot={0}
    dotStyle={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#0084F8",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 3,
        "height": 10,
        "marginHorizontal": 2,
        "width": 10,
      }
    }
    dotsLength={3}
    inactiveDotOpacity={1}
    inactiveDotScale={1}
    inactiveDotStyle={
      Object {
        "backgroundColor": "#0084F8",
        "borderRadius": 5,
        "height": 10,
        "marginHorizontal": 2,
        "width": 10,
      }
    }
    tappableDots={false}
    vertical={false}
  />
</Fragment>
`;
